trie/secure_trie.go

```diff
@@ -37,6 +37,7 @@ import (
 // SecureTrie is not safe for concurrent use.
 type SecureTrie struct {
 	trie             Trie
+	db               *preimageStore
```

Comment:
would you mind renaming this field to something else? `SecureTrie` implements `Trie` which also has a `Database()` method. I have the feeling that naming the preimage store `db` would be a source of great confusion

Reply:
I agree, `preimages` maybe?